### ISSUE TITLE
E265 Block comment should start with `# `

### ISSUE THREAD

**Issue by WhyNotHugo** [astral-sh/ruff#8003]
I have the following code:

```py
#! Keys in the dict are products (OrderItem)
#! Values are (amount, discount)
Discounts = dict[OrderItem, tuple[int, int]]
```

Ruff (in preview mode) complains about these:

```console
> ruff check --preview .
idf/core/models/__init__.py:2743:1: E265 Block comment should start with `# `
idf/core/models/__init__.py:2744:1: E265 Block comment should start with `# `
```

The `#!` syntax is use to document members like these (both variables and attributes). It's picked up by Sphinx and rendered similar to a docstring.

While I understand the purpose of the lint (e.g.: to ensure the leading space), I think this particular syntax should be kept as an exception.

**Comment by charliermarsh** [comment]
Makes sense. Pycodestyle flags this too, but we can avoid it.

**Comment by WhyNotHugo** [comment]
My bad, I meant to say `#:`. Looks like the sample that I was originally looking at was actually wrong.

**Comment by WhyNotHugo** [comment]
I see that `#:` works fine so closing. Sorry for the noise here.

**Comment by charliermarsh** [comment]
No worries, thanks for following up :)
